share/rocm/cmake/ROCMCreatePackage.cmake

```diff
@@ -274,3 +318,45 @@ macro(rocm_set_comp_cpackvar HEADER_ONLY components)
         set(CPACK_DEBIAN_${COMPONENT}_FILE_NAME "DEB-DEFAULT")
     endforeach()
 endmacro()
+
+macro(rocm_package_setup_component COMPONENT_NAME)
+    set(options)
+    set(oneValueArgs PACKAGE_NAME LIBRARY_NAME PARENT)
+    set(multiValueArgs DEPENDS)
+
+    cmake_parse_arguments(PARSE "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    list(APPEND ROCM_PACKAGE_COMPONENTS ${COMPONENT_NAME})
+
+    if(NOT DEFINED PARSE_PACKAGE_NAME)
+        string(TOLOWER "${COMPONENT_NAME}" PARSE_PACKAGE_NAME)
+        if(NOT BUILD_SHARED_LIBS)
+            set(PARSE_PACKAGE_NAME "${PARSE_PACKAGE_NAME}-static")
+        endif()
+    endif()
+
+    if(NOT DEFINED PARSE_LIBRARY_NAME)
+        set(PARSE_LIBRARY_NAME "${PROJECT_NAME}")
+    endif()
+
+    string(TOUPPER "${COMPONENT_NAME}" COMPONENT_GNAME)
+
+    set(CPACK_DEBIAN_${COMPONENT_GNAME}_PACKAGE_NAME "${PARSE_LIBRARY_NAME}-${PARSE_PACKAGE_NAME}")
+    set(CPACK_RPM_${COMPONENT_GNAME}_PACKAGE_NAME "${PARSE_LIBRARY_NAME}-${PARSE_PACKAGE_NAME}")
+
+    if(DEFINED PARSE_PARENT)
+        list(APPEND ROCM_PACKAGE_COMPONENT_DEPENDENCIES "${PARSE_PARENT}->${COMPONENT_NAME}")
+    endif()
+
+    if(DEFINED PARSE_DEPENDS)
+        cmake_parse_arguments(DEPENDS "RUNTIME" "" "COMMON;DEBIAN;RPM;COMPONENT" ${PARSE_DEPENDS})
```

Comment:
Why cant `RUNTIME` be listed under `COMPONENT`?

Reply:
Another option would be to set `CMAKE_INSTALL_DEFAULT_COMPONENT_NAME` to `runtime` instead of `Unspecified`, then `rocm_package_setup_component([...] DEPENDS COMPONENT runtime [...])` would behave as expected.

Reply:
> Another option would be to set CMAKE_INSTALL_DEFAULT_COMPONENT_NAME to runtime instead of Unspecified

That would make things clearer.

Reply:
This has now been implemented.